src/java.base/share/classes/java/net/URLDecoder.java

```diff
@@ -144,24 +144,28 @@ public static String decode(String s, String enc) throws UnsupportedEncodingExce
      * Decodes an {@code application/x-www-form-urlencoded} string using
      * a specific {@linkplain Charset Charset}.
      * The supplied charset is used to determine
-     * what characters are represented by any consecutive sequences of the
-     * form "<i>{@code %xy}</i>".
+     * what characters are represented by any consecutive escape sequences of
+     * the form "<i>{@code %xy}</i>".
      * <p>
      * <em><strong>Note:</strong> The <a href=
      * "http://www.w3.org/TR/html40/appendix/notes.html#non-ascii-chars">
      * World Wide Web Consortium Recommendation</a> states that
      * UTF-8 should be used. Not doing so may introduce
      * incompatibilities.</em>
+     * <p>
+     * If any consecutive well-formed escape sequences cannot
+     * be decoded as a sequence of characters in the supplied {@code Charset}
+     * {@linkplain java.nio.charset.CharsetDecoder##cae the replacement character} will be used.
```

Comment:
I think it would be a bit clearer to say that erroneous bytes are replaced with the Charset's replacement value.

Reply:
Thanks for the suggestion, I just wanted to make sure I was understanding you correctly before committing the change. 

Would it be something like this?

```      
     * Erroneous bytes are replaced with the supplied {@code Charset}'s
     * {@linkplain java.nio.charset.CharsetDecoder##cae replacement value}.
```

Reply:
I am OK with new text on the condition it is moved inside the paragraph that talks about decoding (appended to lines 147-148 above):

```
     * The supplied charset is used to determine
     * what characters are represented by any consecutive escape sequences of
     * the form "<i>{@code %xy}</i>".   Erroneous bytes are replaced with the 
     * supplied {@code Charset}'s {@linkplain java.nio.charset.CharsetDecoder##cae
     * replacement value}.
```